be1-go/hub/witness_test.go

```diff
@@ -0,0 +1,72 @@
+package hub
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"log"
+	"strings"
+	"student20_pop"
+	"sync"
+	"testing"
+)
+
+func CreateWitnessHub() Hub {
+	pk := "OgFFZz2TVilTSICEdJbAO3otWGfh17SmPo6i5as7XAg="
+	pkBuf, err := base64.URLEncoding.DecodeString(pk)
+	if err != nil {
+		return nil
+	}
+	point := student20_pop.Suite.Point()
+	err = point.UnmarshalBinary(pkBuf)
+	if err != nil {
+		return nil
+	}
+	h, err := NewWitnessHub(point)
+	if err != nil {
+		return nil
+	}
+	return h
+}
+
+// to complete
+func TestNewWitnessHub(t *testing.T) {
+	pk := "invalid pk"
+	_, err := base64.URLEncoding.DecodeString(pk)
+	if err == nil {
+		t.Errorf("decoded invalid string")
+	}
```

Comment:
For the other tests we are using the `require` package. But maybe we can let it like this.